pallet-lo-authority-list/src/tests.rs

```diff
@@ -171,17 +191,135 @@ fn it_updates_nodes_on_update() {
 	});
 }
+
+#[test]
+fn it_adds_guest() {
+	new_test_ext().execute_with(|| {
+		setup_host_and_guest();
+		assert!(LoAuthorityList::legal_officer_set(LEGAL_OFFICER_ID2).is_some());
+		assert_eq!(LoAuthorityList::legal_officer_nodes().len(), 1);
+	});
+}
+
+fn setup_host_and_guest() {
+	let host_data = LegalOfficerData::Host(HostData {
+		node_id: Option::Some(OpaquePeerId(bs58::decode("12D3KooWBmAwcd4PJNJvfV89HwE48nwkRmAgo8Vy3uQEyNNHBox2").into_vec().unwrap())),
+		base_url: Option::Some("https://node.logion.network".as_bytes().to_vec()),
+	});
+	assert_ok!(LoAuthorityList::add_legal_officer(RuntimeOrigin::root(), LEGAL_OFFICER_ID, host_data));
+	assert_ok!(LoAuthorityList::add_legal_officer(RuntimeOrigin::root(), LEGAL_OFFICER_ID2, LegalOfficerData::Guest(LEGAL_OFFICER_ID)));
+}
+
+#[test]
+fn it_removes_guest() {
+	new_test_ext().execute_with(|| {
+		setup_host_and_guest();
+		assert_ok!(LoAuthorityList::remove_legal_officer(RuntimeOrigin::root(), LEGAL_OFFICER_ID2));
+		assert!(LoAuthorityList::legal_officer_set(LEGAL_OFFICER_ID2).is_none());
+		assert_eq!(LoAuthorityList::legal_officer_nodes().len(), 1);
+	});
+}
+
+#[test]
+fn it_turns_guest_into_host() {
+	new_test_ext().execute_with(|| {
+		setup_host_and_guest();
+		let host_data2 = LegalOfficerData::Host(HostData {
+			node_id: Option::Some(OpaquePeerId(bs58::decode("12D3KooWQYV9dGMFoRzNStwpXztXaBUjtPqi6aU76ZgUriHhKust").into_vec().unwrap())),
+			base_url: Option::Some("https://node2.logion.network".as_bytes().to_vec()),
+		});
+		assert_ok!(LoAuthorityList::update_legal_officer(RuntimeOrigin::root(), LEGAL_OFFICER_ID2, host_data2));
+		assert!(LoAuthorityList::legal_officer_set(LEGAL_OFFICER_ID2).is_some());
+		assert_eq!(LoAuthorityList::legal_officer_nodes().len(), 2);
+	});
+}
+
+#[test]
+fn it_turns_host_into_guest() {
+	new_test_ext().execute_with(|| {
+		setup_hosts();
+		assert_ok!(LoAuthorityList::update_legal_officer(RuntimeOrigin::root(), LEGAL_OFFICER_ID2, LegalOfficerData::Guest(LEGAL_OFFICER_ID)));
+		assert!(LoAuthorityList::legal_officer_set(LEGAL_OFFICER_ID2).is_some());
+		assert_eq!(LoAuthorityList::legal_officer_nodes().len(), 1);
+	});
+}
+
+fn setup_hosts() {
+	let host_data = LegalOfficerData::Host(HostData {
+		node_id: Option::Some(OpaquePeerId(bs58::decode("12D3KooWBmAwcd4PJNJvfV89HwE48nwkRmAgo8Vy3uQEyNNHBox2").into_vec().unwrap())),
+		base_url: Option::Some("https://node1.logion.network".as_bytes().to_vec()),
+	});
+	assert_ok!(LoAuthorityList::add_legal_officer(RuntimeOrigin::root(), LEGAL_OFFICER_ID, host_data));
+	let host_data2 = LegalOfficerData::Host(HostData {
+		node_id: Option::Some(OpaquePeerId(bs58::decode("12D3KooWQYV9dGMFoRzNStwpXztXaBUjtPqi6aU76ZgUriHhKust").into_vec().unwrap())),
+		base_url: Option::Some("https://node2.logion.network".as_bytes().to_vec()),
+	});
+	assert_ok!(LoAuthorityList::add_legal_officer(RuntimeOrigin::root(), LEGAL_OFFICER_ID2, host_data2));
+}
+
+#[test]
+fn it_fails_turning_host_with_guest_into_guest() {
```

Comment:
Just to be complete, is it possible to also test against error `HostCannotConvert` ?

Reply:
Of course.